compiler/quilt/tools/package.py

```diff
@@ -229,7 +229,11 @@ def save_file(self, srcfile, name, path):
         self._add_to_contents(fullname, [filehash], '', path, 'file', None)
         objpath = self._store.object_path(filehash)
         if not os.path.exists(objpath):
-            copyfile(srcfile, objpath)
+            # Copy the file to a temporary location first, then move, to make sure we don't end up with
+            # truncated contents if the build gets interrupted.
+            tmppath = self._store.temporary_object_path(filehash)
+            copyfile(srcfile, tmppath)
+            move(tmppath, objpath)
```

Comment:
performance issue for large-sized packages? (or many files...)

Reply:
It's just a move, though - so doesn't involve any copying. (The temp dir and the real one are both inside of the `quilt_packages` dir, so same file system.) The point of using `move` is that it should be quick and atomic.